app/controllers/posts_controller.rb

```diff
@@ -1,29 +1,25 @@
 class PostsController < ApplicationController
-  before_action :set_post, only: [:show, :edit, :destroy]
+  before_action :set_post, only: %i[show edit destroy]
+  before_action :authenticate_user! , only:[:new , :create]
 
   # GET /posts
   # GET /posts.json
   def index
     @posts = Post.all
-    if user_signed_in?
-    @post = current_user.posts.build
-    end
+    @post = current_user.posts.build if user_signed_in?
     @users = User.all
   end
 
   # GET /posts/1
   # GET /posts/1.json
-  def show
-  end
+  def show; end
 
   # GET /posts/new
-  def new
-    @post = current_user.posts.build
+  def new 
```

Comment:
Create a new `Post` object here with `@post = Post.new`.
[Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/authentication/posts_new_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/authentication) :
![image](https://user-images.githubusercontent.com/2739245/95663317-e2a1e400-0b03-11eb-8e83-3edeff949ceb.png)

Reply:
I get your point, put, posts are created on my index page and use the index action, so I have to create a new post instance every time I open the page, so there is no use of having a new action at all.

Reply:
@mahmoud717, sorry to not have replied earlier.

The point here is not just cosmetic, it's a matter of good practice. 

A thorough explanation of why this is done in this way can be found here: https://stackoverflow.com/questions/5249151/what-is-the-purpose-of-the-new-controller-action-in-rails

It's a matter of maintainability of the code.

Hope this helps.

Regards!